app/code/Magento/Widget/Model/Widget.php

```diff
@@ -456,4 +460,65 @@ protected function sortParameters($firstElement, $secondElement)
         $bOrder = (int)$secondElement->getData('sort_order');
         return $aOrder < $bOrder ? -1 : ($aOrder > $bOrder ? 1 : 0);
     }
+
+    /**
+     * Encode reserved chars
+     *
+     * @param string $string
+     * @return string|string[]
+     */
+    private function encodeReservedChars($string)
+    {
+        $map = [
+            '{' => urlencode('{'),
+            '}' => urlencode('}')
+        ];
```

Comment:
I guess it would be better to have this as internal property. If we need to add a new char to this we would have to add it to encode and decode functions, what do you think?